tests/jax/models/mnist/cnn/model_implementation.py

```diff
@@ -0,0 +1,32 @@
+# SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import jax
+from flax import linen as nn
+
+
+class MNISTCNNModel(nn.Module):
+    @nn.compact
+    def __call__(self, x, *, train: bool):
+        x = nn.Conv(features=32, kernel_size=(3, 3), padding="SAME")(x)
+        x = nn.BatchNorm(use_running_average=not train, epsilon=1e-5)(x)
+        x = nn.relu(x)
+        x = nn.max_pool(x, window_shape=(2, 2), strides=(2, 2))
+
+        x = nn.Conv(features=64, kernel_size=(3, 3), padding="SAME")(x)
+        x = nn.BatchNorm(use_running_average=not train, epsilon=1e-5)(x)
+        x = nn.relu(x)
+        x = nn.max_pool(x, window_shape=(2, 2), strides=(2, 2))
+
+        x = x.reshape((x.shape[0], -1))
+        x = nn.Dense(features=256)(x)
+        x = nn.BatchNorm(use_running_average=not train, epsilon=1e-5)(x)
+        x = nn.relu(x)
+
+        x = nn.Dense(features=128)(x)
+        x = nn.BatchNorm(use_running_average=not train, epsilon=1e-5)(x)
+        x = nn.relu(x)
+        x = nn.Dense(features=10)(x)
+        x = nn.softmax(x)
```

Comment:
I would add blank lines above and below these two lines to separate the final fc layer too, similar to other layers